flutter_cache_manager/lib/src/web/web_helper.dart

```diff
@@ -106,11 +106,6 @@ class WebHelper {
       headers.addAll(authHeaders);
     }
 
-    final etag = cacheObject.eTag;
-    if (etag != null) {
-      headers[HttpHeaders.ifNoneMatchHeader] = etag;
-    }
```

Comment:
```suggestion
    // Adding `if-none-match` header on web causes a CORS error.
    if (etag != null && !kIsWeb) { 
      headers[HttpHeaders.ifNoneMatchHeader] = etag;
    }
```

Or we could go about it this way...

Reply:
At least on Android and iOS we need to handle the eTag ourselves. I didn't know that browsers handle the eTag for you even for normal http calls. We do need to keep this on everything but the web, so the proposed solution with `&& !kIsWeb` is better IMO.